hnn_core/network.py

```diff
@@ -741,6 +780,8 @@ def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
                     weights_by_type.update({target_type: {'ampa': 0.}})
                 if target_type not in delays_by_type:
                     delays_by_type.update({target_type: 0.1})
+                if target_type not in probability_by_type:
+                    probability_by_type.update({target_type: 1.0})
```

Comment:
If the user doesn't specify the connection probability for a certain cell type, should the default legacy mode behavior be to set the probability at 1.0 or 0?

Reply:
I think it should default to 1.0 since that is the default when you don't specify `probability` at all